app/test/unit/parse-carriage-return-test.ts

```diff
@@ -0,0 +1,111 @@
+      '\r' +
+      '                                                                                \r' +
+      'Receiving objects:  37% (3/8)\r' +
+      '\r' +
+      '                                                                                \r' +
+      'Receiving objects:  50% (4/8)\r' +
+      '\r' +
+      '                                                                                \r' +
+      'Receiving objects:  62% (5/8)\r' +
+      '\r' +
+      '                                                                                \r' +
+      'Receiving objects:  75% (6/8)\r' +
+      '\r' +
+      '                                                                                \r' +
+      'Receiving objects:  87% (7/8)\r' +
+      '\r' +
+      '                                                                                \r' +
+      'Receiving objects: 100% (8/8)\r' +
+      '\r' +
+      '                                                                                \r' +
+      'Receiving objects: 100% (8/8), done.\n' +
+      '\r' +
+      '                                                                                \r' +
+      'Resolving deltas:   0% (0/1)\r' +
+      '\r' +
+      '                                                                                \r' +
+      'Resolving deltas: 100% (1/1)\r' +
+      '\r' +
+      '                                                                                \r' +
+      'Resolving deltas: 100% (1/1), done.\n' +
+      "fatal: multiple updates for ref 'refs/remotes/origin/pr/1' not allowed\n"
+
+    const expected =
+      "Cloning into '/Users/markus/Documents/GitHub/delete-branch-test'...\n" +
+      'remote: Enumerating objects: 8, done.        \n' +
+      'remote: Counting objects: 100% (8/8), done.        \n' +
+      'remote: Compressing objects: 100% (6/6), done.        \n' +
+      'remote: Total 8 (delta 1), reused 7 (delta 0), pack-reused 0        \n' +
+      'Receiving objects: 100% (8/8), done.                                            \n' +
+      'Resolving deltas: 100% (1/1), done.                                             \n' +
+      "fatal: multiple updates for ref 'refs/remotes/origin/pr/1' not allowed\n"
+
+    expect(parseCarriageReturn(cloneOutput)).toBe(expected)
+  })
+
+  it("has no problems with strings that don't contain carriage returns", () => {
+    expect(parseCarriageReturn('foo')).toBe('foo')
+    expect(parseCarriageReturn('foo\nbar')).toBe('foo\nbar')
+  })
+
+  it('handles terminating carriage returns', () => {
+    expect(parseCarriageReturn('foo\r')).toBe('foo')
+    expect(parseCarriageReturn('foo\nbar\r')).toBe('foo\nbar')
+  })
+
+  it('handles strings terminating without newline or CR', () => {
+    expect(parseCarriageReturn('foo\rbar')).toBe('bar')
+    expect(parseCarriageReturn('foo\r\r\r\r\rbar')).toBe('bar')
+  })
+
+  it('treats unicode line separator and paragraph separator as flow text', () => {
```

Comment:
@rafeca This test is kinda pointless now that we're not using a regex but I think it's a fair test still in case we alter course in the future and turn it into a regex again. I'll drop it if you want though